code/modules/surgery/surgery_step.dm

```diff
@@ -211,3 +211,18 @@
 	user.visible_message(detailed_message, self_message, vision_distance = 1, ignored_mobs = target_detailed ? null : target)
 	if(!target_detailed)
 		to_chat(target, vague_message)
+
+/**
+ * Sends a pain message to the target, including a chance of screaming.
+ *
+ * Arguments:
+ * * target - Who the message will be sent to
+ * * pain_message - The message to be displayed
+ * * mechanical_surgery - Boolean flag that represents if a surgery step is done on a mechanical limb (therefore does not force scream)
+ */
+/datum/surgery_step/proc/display_pain(mob/living/target, pain_message, mechanical_surgery = FALSE)
+	if(target.stat < UNCONSCIOUS)
+		to_chat(target, "<span class='userdanger'>[pain_message]</span>")
+		SEND_SIGNAL(target, COMSIG_ADD_MOOD_EVENT, "surgery", /datum/mood_event/surgery)
+		if(prob(30) && !mechanical_surgery)
+			target.emote("scream")
```

Comment:
If the limb is mechanical, then should it cause any pain to operate on at all?

Reply:
At least I don't think it should apply the same mood debuff

Reply:
Applying the same mood debuff will only refresh its duration, no worries.